build.gradle.kts

```diff
@@ -102,3 +105,80 @@ tasks.register("mergeDashFiles") {
         }
     }
 }
+
+// region jacoco coverage report
+
+subprojects {
+    apply {
+        plugin("jacoco")
+    }
+
+    if (plugins.hasPlugin("com.android.library")) {
+        configure<LibraryExtension> {
+            @Suppress("UnstableApiUsage")
+            testOptions {
+                buildTypes {
+                    named("debug") {
+                        enableUnitTestCoverage = true
+                        enableAndroidTestCoverage = true
+                    }
+                }
+            }
+        }
+    }
+
+    if (plugins.hasPlugin("com.android.application")) {
+        configure<BaseAppModuleExtension> {
+            @Suppress("UnstableApiUsage")
+            testOptions {
+                buildTypes {
+                    named("debug") {
+                        enableUnitTestCoverage = true
+                        enableAndroidTestCoverage = true
+                    }
+                }
+            }
+        }
+    }
+}
+
+tasks.create("jacocoRootReport", JacocoReport::class.java) {
+    group = "report"
+
+    reports {
+        html.required.set(true)
+        xml.required.set(true)
+        csv.required.set(false)
+    }
+
+    val excludes = listOf(
+        "**/buildSrc/**",
+        "**/app/**",
+        "**/samples/**",
+        "**/build/**/org/eclipse/kuksa/vss/**", // generated
+        "**/test*/**/*.*",
+    )
+
+    val sourcesKotlin = subprojects.map { it.layout.projectDirectory.dir("src/main/kotlin") }
+    val sourcesJava = subprojects.map { it.layout.projectDirectory.dir("src/main/java") }
```

Comment:
I intentionally left the vss-processor-plugin out-of-scope because it does not add any coverage. Check for yourself: 

[vss-plugin-code-coverage.diff.txt](https://github.com/eclipse-kuksa/kuksa-android-sdk/files/14449042/vss-plugin-code-coverage.diff.txt)

test.exec, sources and classes are correctly provided, still it doesn't show. If you add the test.exec in Android Studio you can see the following coverage data being generated: 
![grafik](https://github.com/eclipse-kuksa/kuksa-android-sdk/assets/138565883/52201c0b-5210-42d0-8cfb-320c2a51ef17)

I can still add it, but personally don't think it's useful. Your decision.

Reply:
'\*\*/vssprocessor/plugin/\*\*' was excluded. A follow-up ticket was created: #91